simulator.py

```diff
@@ -0,0 +1,41 @@
+import numpy
+
+
+class Simulator(object):
+    def __init__(self, state_space, action_space, discount):
+        self.state_space = state_space
+        self.action_space = action_space
+        self.discount = discount
+        self.max_iterations = 1000
+        self.tolerance = 0.01
+
+    def find_best_policy_for_state(self, state, value_func):
+        best_action = None
+        best_payoff = float('-inf')
+        for a, payoff in self.action_space.get_actions(state):
+            for next_state, prob in self.action_space.transition_probabilities(a, state):
+                payoff += value_func[next_state] * prob * self.discount
```

Comment:
Isn't it common to accumulate values in a for loop?  Do you suggest using accumulate instead?